### ISSUE TITLE
#66 persists for me in portable mode, even though I have 2.9.6942

### ISSUE THREAD

**Issue by georgerbr** [MiKTeX/miktex#289]
This issue persists for me, even though I have a fully updated installation:

I installed miktex-portable-2.9.6942 and added my local directories - as soon as I closed the console and opened it again, the directory entries would be lost. I had to find and install an old copy of `miktexstartup.ini` for my local directories to work:

```
;;; MiKTeX startup configuration


[Auto]
Config=Portable

[Paths]

;; ..\..\..\..\texmf;..\..\..\..\texmf-paid
UserRoots=
UserRoots;=..\..\..\..\texmf
UserRoots;=..\..\..\..\texmf-paid
```

_Originally posted by @georgerbr in https://github.com/MiKTeX/miktex/issues/66#issuecomment-398396097_

**Comment by edocevoli** [comment]
Please follow the [contribution guidelines](https://github.com/MiKTeX/miktex/blob/master/CONTRIBUTING.md) when submitting an issue ticket.

**Comment by cyrildtm** [comment]
Here are the details. 

Check list

Please review this check list, before you click the Submit button:

1.     the bug report is self-contained: it does not contain links to questions/reports on external sites (e.g., Stack Exchange Network)
This is a repeated appearance of Issue #66. 

2.     step-by-step reproduction instructions

Step 1: Follow the instructions at https://miktex.org/kb/texmf-roots, which are: 
    Step 1.1: Create a directory D:\typewrite\library which serves as the new TEXMF root directory.
    Step 1.2: Create TDS-compliant subdirectories and move/copy your files here. In the folder I have tex\latex\mystyle which contains mystyle.sty
    Step 1.3: Start MiKTeX Console and open the Settings page
    Step 1.4: Click the Directories tab.
    Step 1.5: Click the Add button and browse to the TEXMF root directory created earlier (step 1). I have D:\typewrite\library

Step 2: Wait 30 seconds. Then close MiKTeX Console completely. 
Step 3: Start MiKTeX Console and open the Settings page
Step 4: Click the Directories tab.
Step 5: Observe the TEXMF root directories list, and it does not contain D:\typewrite\library

3.     configuration report (initexmf --report) attached
Date: 2019-04-18 15:12:44
MiKTeX: 2.9.7000
OS: Windows 7 Enterprise, 64-bit, Service Pack 1, build 7601
SharedSetup: no
LinkTargetDirectory: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\te
xmfs\install\miktex\bin
PathOkay: yes
LastUpdateCheck: not yet
LastUpdate: not yet
SystemAdmin: yes
RootPrivileges: no
AdminMode: no
Root0: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\config
Root1: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\data
Root2: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\install
UserInstall: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\ins
tall
UserConfig: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\conf
ig
UserData: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\data
CommonInstall: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\i
nstall
CommonConfig: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\co
nfig
CommonData: C:\Users\User\Documents\latex\miktex-portable-2.9.6942\texmfs\data

4.     input files attached
There is no input file involved. 
5.     log files attached

[logfiles.zip](https://github.com/MiKTeX/miktex/files/3095868/logfiles.zip)

**Comment by cyrildtm** [comment]
More information: 

Running the following commands doesn't help, exactly the same symptom.

    `initexmf --user-roots="path\to\library" initexmf --update-fndb`



I solved the problem by manually editing this file in the miktex-portable folder: `texmfs\install\miktex\config\miktexstartup.ini`

which only contains a `[Auto]` field. I manually add a `[Path]` field containing the following line

`UserData=path\to\library`

There are white spaces in my library path, and I did not do anything about it like any quotation mark. I do not put any white space around the equal sign.

I reboot console and my tex editor (TeXstudio) and it works again.
